common/pkg/auth/authenticator.go

```diff
@@ -39,22 +42,37 @@ type reqIntercepter interface {
 // ExternalAuthenticator authenticates external requests with RBAC server.
 type ExternalAuthenticator struct {
 	intercepter    reqIntercepter
+	rbacClient     rbacv1.RbacInternalServiceClient
 	tokenExchanger *TokenExchanger
+	loginCache     *cache
 }
 
 // NewExternalAuthenticator returns a new ExternalAuthenticator.
-func NewExternalAuthenticator(ctx context.Context, addr string, tex *TokenExchanger) (*ExternalAuthenticator, error) {
+func NewExternalAuthenticator(
+	ctx context.Context,
+	rbacServerAddr string,
+	tex *TokenExchanger,
+	cacheExpiration, cacheCleanup time.Duration,
+) (*ExternalAuthenticator, error) {
 	i, err := auth.NewInterceptor(ctx, auth.Config{
-		RBACServerAddr: addr,
+		RBACServerAddr: rbacServerAddr,
 		// TODO(kenji): Revisit.
 		AccessResource: "api.fine_tuning.jobs",
 	})
 	if err != nil {
 		return nil, err
 	}
+
+	conn, err := grpc.DialContext(ctx, rbacServerAddr, grpc.WithTransportCredentials(insecure.NewCredentials()))
```

Comment:
nitpick: `DianlCotext` is deprecated. How about `NewClient`?

https://pkg.go.dev/google.golang.org/grpc#DialContext